bleak/backends/corebluetooth/CentralManagerDelegate.py

```diff
@@ -61,22 +60,128 @@
 DisconnectCallback = Callable[[], None]
 
 
-class CentralManagerDelegate(NSObject):
+class ObjcCentralManagerDelegate(NSObject):
+    """macOS conforming python class for managing the CentralManger for BLE"""
+
+    ___pyobjc_protocols__ = [CBCentralManagerDelegate]
+
+    def initWithPyDelegate_(
+        self, py_delegate: CentralManagerDelegate
+    ) -> Optional[ObjcCentralManagerDelegate]:
```

Comment:
```suggestion
    ) -> Optional[Self]:
```

I know we are never going to subclass this, but `Self` is still technically correct.